src/server/routes/register.ts

```diff
@@ -110,6 +133,262 @@ router.post(
 	}),
 );
 
+// Essentially the email-sent page, but for passcode registration
+router.get('/register/code', (req: Request, res: ResponseWithRequestState) => {
+	const state = res.locals;
+
+	const encryptedState = readEncryptedStateCookie(req);
+
+	if (encryptedState && encryptedState.email && encryptedState.stateHandle) {
+		const html = renderer('/register/email-sent', {
+			requestState: mergeRequestState(state, {
+				pageData: {
+					email: readEncryptedStateCookie(req)?.email,
+					timeUntilTokenExpiry: encryptedState.stateHandleExpiresAt
```

Comment:
You could be pedantic/avoid confusion and call this `msUntilTokenExpiry`. Just because some OAuth stuff uses seconds instead.

Reply:
I did it this way as `timeUntilTokenExpiry` was already a thing in Gateway, and wanted to avoid a refactor.

Reply:
hopefully the new function name makes it clearer